### PR TITLE
reference/configuration: update tidb-server configuration

### PR DESCRIPTION
**Pull request by Yisaer**
### What is changed, added or deleted? (Required)
add new configuration in tidb-server configuration document

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

https://github.com/pingcap/tidb/pull/15700

### CODE REVIEW

File: reference/configuration/tidb-server/configuration-file.md [hunk 1]
```diff
@@ -33,6 +33,14 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：`<操作系统临时文件夹>/tidb/tmp-storage`
 + 此配置仅在 `oom-use-tmp-storage` 为 true 时有效。
 
+### `temp-storage-quota`
+
++ 当 `oom-use-tmp-storage` 开启时，对于 `tmp-storage-path` 存储使用的限额，单位为字节。
++ 当 `tmp-storage-path` 的容量低于 `temp-storage-quota` 所定义的值时，tidb-server 启动时将会报出错误并退出
```
Comment (SunRunAway) [inline reference/configuration/tidb-server/configuration-file.md:39]: `tmp-storage-path` 的容量
这个是指总容量还是剩余可用容量？

File: reference/configuration/tidb-server/configuration-file.md [hunk 2]
```diff
@@ -33,6 +33,14 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
++ 当某些算子在使用临时磁盘时，tidb-server 的临时磁盘存储总是用量超过 `temp-storage-quota` 时，将会取消当前 SQL 操作，并返回 `Out Of Global Storage Quota!`
```
Comment (SunRunAway) [inline reference/configuration/tidb-server/configuration-file.md:40]:
```suggestion
+ 当单条 SQL 语句使用临时磁盘，导致 tidb-server 的总体临时磁盘总量超过 `temp-storage-quota` 时，将会取消当前 SQL 操作，并返回 `Out Of Global Storage Quota!`
```

File: reference/configuration/tidb-server/configuration-file.md (hunk 1 above)
Comment (Yisaer) [reply reference/configuration/tidb-server/configuration-file.md]: 剩余可用容量，updated.

File: reference/configuration/tidb-server/configuration-file.md [hunk 3]
```diff
@@ -33,6 +33,14 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：`<操作系统临时文件夹>/tidb/tmp-storage`
 + 此配置仅在 `oom-use-tmp-storage` 为 true 时有效。
 
+### `temp-storage-quota`
+
++ 当 `oom-use-tmp-storage` 开启时，对于 `tmp-storage-path` 存储使用的限额，单位为字节。
```
Comment (SunRunAway) [inline reference/configuration/tidb-server/configuration-file.md:38]: 我觉得前半句不用了。 不然这句信息太多。
cc @zz-jason

File: reference/configuration/tidb-server/configuration-file.md [hunk 4]
```diff
@@ -33,6 +33,14 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
++ 当 `tmp-storage-path` 的剩余可用容量低于 `temp-storage-quota` 所定义的值时，tidb-server 启动时将会报出错误并退出。
```
Comment (SunRunAway) [inline reference/configuration/tidb-server/configuration-file.md:39]: 这句可以放到最后

File: reference/configuration/tidb-server/configuration-file.md (hunk 3 above)
Comment (Yisaer) [reply reference/configuration/tidb-server/configuration-file.md]: updated.

File: reference/configuration/tidb-server/configuration-file.md (hunk 4 above)
Comment (Yisaer) [reply reference/configuration/tidb-server/configuration-file.md]: updated.

File: reference/configuration/tidb-server/configuration-file.md [hunk 5]
```diff
@@ -33,6 +33,14 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：`<操作系统临时文件夹>/tidb/tmp-storage`
 + 此配置仅在 `oom-use-tmp-storage` 为 true 时有效。
 
+### `temp-storage-quota`
+
++ 对于 `tmp-storage-path` 存储使用的限额，单位为字节。
```
Comment (TomShawn) [inline reference/configuration/tidb-server/configuration-file.md:38]:
```suggestion
+ `tmp-storage-path` 存储使用的限额，单位为字节。
```

File: reference/configuration/tidb-server/configuration-file.md [hunk 6]
```diff
@@ -33,6 +33,14 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
++ 当单条 SQL 语句使用临时磁盘，导致 tidb-server 的总体临时磁盘总量超过 `temp-storage-quota` 时，将会取消当前 SQL 操作，并返回 `Out Of Global Storage Quota!`
```
Comment (TomShawn) [inline reference/configuration/tidb-server/configuration-file.md:39]:
```suggestion
+ 当单条 SQL 语句使用临时磁盘，导致 TiDB server 的总体临时磁盘总量超过 `temp-storage-quota` 时，当前 SQL 操作会被取消，并返回 `Out Of Global Storage Quota!` 错误。
```

File: reference/configuration/tidb-server/configuration-file.md [hunk 7]
```diff
@@ -33,6 +33,14 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
++ 当 `temp-storage-quota` 小于 0 时则没有上述检查与限制。
++ 默认值: -1
++ 当 `tmp-storage-path` 的剩余可用容量低于 `temp-storage-quota` 所定义的值时，tidb-server 启动时将会报出错误并退出。
```
Comment (TomShawn) [inline reference/configuration/tidb-server/configuration-file.md:42]:
```suggestion
+ 当 `tmp-storage-path` 的剩余可用容量低于 `temp-storage-quota` 所定义的值时，TiDB server 启动时将会报出错误并退出。
```

**Pull request merged**